src/osd/PrimaryLogPG.cc

```diff
@@ -14933,6 +14943,9 @@ void PrimaryLogPG::scrub_snapshot_metadata(
 			     << " : is missing in clone_overlap";
 	  ++scrubber.shallow_errors;
 	  soid_error.set_size_mismatch();
+	} else if (oi->is_evicted_snap()) {
+	  // deduped object evicted
+	  dout(20) << soid << " is evicted manifest object " << dendl;
```

Comment:
This probably can avoid mismatch. What do you think?

Reply:
I don't think this is the right fix.  I think that object_info_t::size should match the *logical* size of the object, not the ondisk size.  You should probably fix that instead and update the portion which checks the ondisk size against the object_info_t::size to ignore a mismatch if it's evicted.